corehq/apps/reports/filters/controllers.py

```diff
@@ -190,20 +190,38 @@ def data_sources(self):
                 (self.get_all_users_size, self.get_all_users),
             ]
 
-    def get_options(self):
-        page = int(self.request.GET.get('page', 1))
-        size = int(self.request.GET.get('page_limit', DEFAULT_PAGE_LIMIT))
-        start = size * (page - 1)
+    @property
+    @memoized
+    def page(self):
+        if self.request.method == 'POST':
+            return int(self.request.POST.get('page', 1))
+        return int(self.request.GET.get('page', 1))
+
+    @property
+    @memoized
+    def size(self):
+        if self.request.method == 'POST':
+            return int(self.request.POST.get('page_limit', DEFAULT_PAGE_LIMIT))
+        return int(self.request.GET.get('page_limit', DEFAULT_PAGE_LIMIT))
+
+    def get_options(self, show_more=False):
```

Comment:
Would it be accurate to rename `show_more` to something like `is_ajax`? All ajax-based selects will use pagination, right? I don't think it's going to be obvious when to pass `show_more` to people who aren't that fmailiar with select2.